src/assets/js/ledyer-payments.js

```diff
@@ -0,0 +1,274 @@
+jQuery( function ( $ ) {
+    if ( typeof LedyerPaymentsParams === "undefined" ) {
+        return false
+    }
+
+    const gatewayParams = LedyerPaymentsParams
+    const { gatewayId, sessionId } = gatewayParams
+
+    /**
+     * Handles the process of proceeding with Ledyer payment for an order.
+     *
+     * @param {string} orderId - The key of the order.
+     * @param {Object} customerData - The customer data.
+     * @returns {void}
+     */
+    const handleProceedWithLedyer = async ( orderId, customerData ) => {
+        blockUI()
+        try {
+            const authArgs = { customer: { ...customerData }, sessionId }
+            const authResponse = await window.ledyer.payments.api.authorize( authArgs )
+
+            // ... some time will pass while the user is interacting with the dialog
+
+            if ( authResponse ) {
```

Comment:
You have a if here, but no else block to catch any cases where its not a true value? Is the if here needed or could it be removed?